mapper/orm/src/main/java/org/hibernate/search/mapper/orm/session/SearchSessionWritePlan.java

```diff
@@ -6,16 +6,27 @@
  */
 package org.hibernate.search.mapper.orm.session;
 
-// TODO HSEARCH-3069 Also document what happens when a transaction is *not* active
 /**
  * An interface for writing to indexes in the context of an ORM Session.
  * <p>
- * When a transaction is active, operations are queued internally,
- * and are only applied on transaction commit.
+ * This class is stateful: it queues operations internally to apply them at a later time.
  * <p>
- * {@link #process()} and {@link #execute()} allow to control explicitly
- * how operations are applied, without relying on transaction commits,
- * which can be useful when indexing large amounts of data in batches.
+ * When {@link #process()} is called,
+ * or when {@link org.hibernate.search.mapper.orm.cfg.HibernateOrmMapperSettings#AUTOMATIC_INDEXING_STRATEGY automatic indexing is enabled}
```

Comment:
I don't think that the AUTOMATIC_INDEXING_STRATEGY is mandatory here. I would write in case of session flush (explicit, automatic, dirty check, query flush, entity flush).

Reply:
It does matter, since it controls whether automatic indexing is enabled or not.

I think you're confusing the automatic indexing strategy with the synchronization strategy.